### DISCUSSION TITLE
Selectable domain redirect type (temporary/permanent)

### DISCUSSION THREAD

**Discussion by ovidyou** [vercel/vercel#4922] [Ideas] [answered]
When using the [domain redirect](https://vercel.com/blog/redirecting-domains) feature, the redirect is a temporary one (HTTP status code 307), which is fine for many use cases, but when dealing with SEO, you would definitely want to have a permanent redirect (301 or 308).

Would be nice to be able to select the domain redirect type (temporary or permanent). Without this, I am forced now to have a separate deployment specifically for doing the permanent redirect, which is not ideal because the deployment seems to become totally unresponsive at times (probably because there is no traffic for longer periods of time).

**Comment by styfle** [comment]
Thanks for the suggestion! We are tracking this feature request internally as Story 3699.

**Reply by lqze** [reply]
Any update on this?

**Reply by krystianjj** [reply]
A 307 is a temporary redirect in the HTTP protocol 1.1. Search Engine bots treat them as 302 redirect. So it is a bad idea ... I wait too :)

**Comment by bigsee** [comment]
@styfle Where could I add a +1 for this one? Presented with this challenge now. Is there any workaround in the meantime?

**Reply by ovidyou** [reply]
@bigsee here is my workaround:

I needed to redirect from `www.example.com` to `example.com`, so I added the `www.example.com` as subdomain in Vercel and assigned a separate repository branch to it. In this branch I have a very basic Next.js app with pages `index.js` and `[...slug].js`, with the sole purpose to manage the redirects. Check the following gist to see how the redirects are managed: https://gist.github.com/ovidyou/b880ca46547b8091e44085639fff29d3

**Reply by ovidyou** [reply]
Another workaround would be to use `next.config.js`:
https://nextjs.org/docs/api-reference/next.config.js/redirects

Because you can only match absolute paths, you would also need a separate deployment branch specifically for the domain or subdomain that you are trying to redirect from. Also you would need Next.js 9.5+ for the redirects configuration.

**Reply by bigsee** [reply]
@ovidyou thank you for sharing this workaround. I am actually working with Nuxt and I _think_ I have this working by doing something similar to your first workaround. I believe it was the workaround from the old Now platform.

I have a second project called `project-name-redirect`. That project has only one file, which is the following now.json file:

```
{
  "version": 2,
  "alias": ["www.project.co.uk"],
  "routes": [
    {
      "src": "/(.*)",
      "status": 301,
      "headers": { "Location": "https://project.co.uk/$1" }
    }
  ]
}
```

I should probably update the filename to vercel.json.

@styfle is this approach also valid/recommended as a workaround?

**Reply by ovidyou** [reply]
@bigsee thanks for sharing that. My first approach was also a now.json configuration, but it didn't work properly:
```
{
  "routes": [
    {
      "src": "/(.*)",
      "status": 301,
      "dest": "https://example.com/$1"
    }
  ]
}
```
For a page like `www.example.com/foo` I got always redirected to `example.com/$1`, so the replacement variable didn't actually get replaced in my case. Don't remember if I also tried with the `Location` header.

**Reply by bigsee** [reply]
@ovidyou 👍🏽 

yeah, try with Location header maybe. It's working this end so far. Good luck!

**Comment by timothyis** [comment, marked as answer]
Hi all! Just an update here to let you know that we've now launched this feature! Read more on our changelog: https://vercel.com/changelog/domains-can-now-be-redirected-with-a-custom-status-code
